vpr/src/noc/noc_traffic_flows.h

```diff
@@ -0,0 +1,307 @@
+
+    // setters
+
+    /**
+     * @brief Given a set of parameters that describe a traffic
+     * flow, create it an add it to the vector of traffic flows in the
+     * design. Additionally, the two router blocks involved have their
+     * ids stored to to keep track of all router blocks that are used
+     * in traffic flows. Finally, the newly created traffic flow is
+     * also added to a vector of traffic flows that are associated to both 
+     * the source and sink routers of the flow.
+     * 
+     * @param source_router_module_name A string that represents the
+     * name of the source router block in the traffic flow. THis is
+     * provided by the user.
+     * @param sink_router_module_name A string that represents the name
+     * of the sink router block in the traffic flow. This is provided by
+     * the user.
+     * @param source_router_cluster_id The source router block id that
+     * uniquely identifies this block in the clustered netlist.
+     * @param sink_router_cluster_id  The sink router block id that
+     * uniquely identifier this block in the clusterd netlist.
+     * @param traffic_flow_bandwidth The size of the data transmission
+     * in this traffic flow (units of bps).
+     * @param traffic_flow_latency The maximum allowable delay between
+     * transmitting data at the source router and having it received
+     * at the sink router.
+     */
+    void create_noc_traffic_flow(std::string source_router_module_name, std::string sink_router_module_name, ClusterBlockId source_router_cluster_id, ClusterBlockId sink_router_cluster_id, double traffic_flow_bandwidth, double traffic_flow_latency);
+
+    /**
+     * @brief Set the status of a given traffic flow. The status indicates
+     * whether the traffic flow has been routed or not.
+     * 
+     * @param traffic_flow_id A unique identifier that represents a traffic
+     * flow.
+     * @param status THe updated status of the traffic flow. True indicating
+     * that the traffic flow has been router and False indicating that
+     * the traffic flow has not been routed.
+     */
+    void set_traffic_flow_status(NocTrafficFlowId traffic_flow_id, bool status);
+
+    //utility functions
+
+    /**
+     * @brief Determines the total number of traffic flows in
+     * the design and creates a vector to keep track of the traffic
+     * flows processed status. This function should be called after
+     * all the traffic flows have been created. The traffic flows are
+     * all initialized to being not processed.
+     * 
+     */
+    void finshed_noc_traffic_flows_setup(void);
+
+    /**
+     * @brief Goes through the status of all traffic flows
+     * and sets them to being not processed. This should be
+     * used before each iteraition of placement.
+     * 
+     */
+    void reset_traffic_flows_processed_status(void);
+
+    /**
+     * @brief Resets the class by clearning internal
+     * satastructures.
```

Comment:
data structures.

Reply:
Think about if / why you need this.  Do you need to clear without destroying?
You're using in the constructor, but do you need to (STL containers come up empty with default construction).

Reply:
But if you think this is useful, keep it.

Reply:
This functionality wont be needed. The traffic flows will only be deleted at the end of the program, so this has been removed.

Reply:
Keeping this for now. I used it within some of the unit tests.